app/scripts/controllers/edit/buildConfig.js

```diff
@@ -414,6 +487,7 @@ angular.module('openshiftConsole')
 
     $scope.save = function() {
       $scope.disableInputs = true;
+      clearIncompatibleBuildHookFields($scope.buildHookSelection.type.id);
```

Comment:
I'd check `$scope.buildHookSelection.type.id` directly in the `clearIncompatibleBuildHookFields` instead of passing it in.